exonum/src/runtime/rust/tests.rs

```diff
@@ -574,10 +600,129 @@ fn state_aggregation() {
     assert_eq!(actual_indexes, expected_indexes);
 }
 
+#[test]
+fn multiple_service_versions() {
+    let runtime = RustRuntime::new(mpsc::channel(1).0)
+        .with_factory(TestServiceImpl)
+        .with_factory(TestServiceImplV2);
+    let config = generate_testnet_config(1, 0)[0].clone();
+    let genesis_config = GenesisConfigBuilder::with_consensus_config(config.consensus).build();
+    let mut blockchain = Blockchain::build_for_tests()
+        .into_mut(genesis_config)
+        .with_runtime(runtime)
+        .build()
+        .unwrap();
+
+    let fork = create_block(&blockchain);
+    let artifact = TestServiceImpl.artifact_id();
+    Dispatcher::commit_artifact(&fork, artifact.clone(), vec![]).unwrap();
+    commit_block(&mut blockchain, fork);
+
+    let mut fork = create_block(&blockchain);
+    let spec = InstanceSpec {
+        artifact: artifact.clone(),
+        id: SERVICE_INSTANCE_ID,
+        name: SERVICE_INSTANCE_NAME.to_owned(),
+    };
+    let constructor = Init { msg: String::new() };
+    ExecutionContext::new(blockchain.dispatcher(), &mut fork, Caller::Blockchain)
+        .start_adding_service(spec, constructor)
+        .unwrap();
+
+    commit_block(&mut blockchain, fork);
+    let snapshot = blockchain.snapshot();
+    let schema = DispatcherSchema::new(&snapshot);
+    assert!(schema.get_artifact(&artifact).is_some());
+    assert!(blockchain.dispatcher().is_artifact_deployed(&artifact));
+
+    // Add a newer artifact version.
+    let fork = blockchain.fork();
+    let new_artifact = TestServiceImplV2.artifact_id();
+    assert_ne!(new_artifact, artifact);
+    assert!(schema.get_artifact(&new_artifact).is_none());
+    Dispatcher::commit_artifact(&fork, new_artifact.clone(), vec![]).unwrap();
+    commit_block(&mut blockchain, fork);
+
+    // ...and a service based on the new artifact.
+    let mut fork = blockchain.fork();
+    let spec = InstanceSpec {
+        artifact: new_artifact.clone(),
+        id: SERVICE_INSTANCE_ID + 1,
```

Comment:
`SERVICE_INSTANCE_ID + 1` and `instance_id += 1` look a bit fragile and hacky, maybe introduce a separate constant for it?